pkg/artifacts/artifact_storage_test.go

```diff
@@ -165,6 +256,132 @@ func TestInitializeArtifactStorageWithConfigMap(t *testing.T) {
 	}
 }
 
+func TestCleanupArtifactStorage(t *testing.T) {
+	logger := logtesting.TestLogger(t)
+	for _, c := range []struct {
+		desc        string
+		configMap   *corev1.ConfigMap
+		pipelinerun *v1alpha1.PipelineRun
+		hasPVC      bool
+	}{{
+		desc: "valid bucket",
+		configMap: &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: system.GetNamespace(),
+				Name:      v1alpha1.BucketConfigName,
+			},
+			Data: map[string]string{
+				v1alpha1.BucketLocationKey:              "gs://fake-bucket",
+				v1alpha1.BucketServiceAccountSecretName: "secret1",
+				v1alpha1.BucketServiceAccountSecretKey:  "sakey",
+			},
+		},
+		pipelinerun: &v1alpha1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "foo",
+				Name:      "pipelineruntest",
+			},
+		},
+		hasPVC: false,
```

Comment:
im a bit confused about what `hasPVC` means - maybe it would be clearer to split this into two separate functions completely, one dealing with all the `hasPVC` cases, and one dealing with the `hasPVC: false` cases (which I think is used to indicate that the configmap configuration is wrong, so there is no PVC to be created?)

Reply:
I just copied in a bunch of test cases from existing tests. Really, we probably don't need the cases where `hasPVC` is false, so I'll remove them.